### PR TITLE
feature: dashboard page numbers

### PR DESCRIPTION
**Pull request by RobCattonKPMG**

### CODE REVIEW

File: packages/siri-sx-generator/test/siriSXUploadToDynamo.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
-import { PutCommand } from "@aws-sdk/lib-dynamodb";
+/* eslint-disable no-console */
```
Comment (alainajaffer) [inline packages/siri-sx-generator/test/siriSXUploadToDynamo.ts:1]: why do we have this for the whole file?

Comment (RobCattonKPMG) [reply packages/siri-sx-generator/test/siriSXUploadToDynamo.ts:1]: There are 2 console.logs - ill remove the inline ones.

File: site/components/PageNumbers.test.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    it("should render correctly when there are 10 pages and the current page is 1", () => {
+        const tree = renderer
+            .create(<PageNumbers currentPage={1} numberOfPages={10} setCurrentPage={vi.fn()} />)
+            .toJSON();
+        expect(tree).toMatchSnapshot();
+    });
+
+    it("should render correctly when there are 10 pages and the current page is 5", () => {
+        const tree = renderer
+            .create(<PageNumbers currentPage={1} numberOfPages={10} setCurrentPage={vi.fn()} />)
```
Comment (alvak1) [inline site/components/PageNumbers.test.tsx:29]: Should the current page be set to 5

File: packages/siri-sx-generator/test/siriSXUploadToDynamo.ts [hunk 3]
```diff
@@ -1,38 +1,105 @@
...
+import { BatchWriteCommand } from "@aws-sdk/lib-dynamodb";
 import { randomUUID } from "crypto";
-import { testDisruptionsJson } from "./testData";
+import { baseDisruptionJson } from "./testData";
 import { getDdbDocumentClient } from "../util/awsClient";
 
-const [stageName] = process.argv.slice(2);
+const [stageName, itemsToCreate] = process.argv.slice(2);
```
Comment (alvak1) [inline packages/siri-sx-generator/test/siriSXUploadToDynamo.ts:7]: The value for `itemsToCreate` is not configured to be passed from the script `publishJsonToDynamo.sh`

Comment (RobCattonKPMG) [reply packages/siri-sx-generator/test/siriSXUploadToDynamo.ts:7]: It is - usage is like this: 

npm run dynamo-publish-json -- rob 100

File: site/components/PageNumbers.test.tsx (hunk 2 above)
Comment (RobCattonKPMG) [reply site/components/PageNumbers.test.tsx:29]: Good catch - will change

File: packages/siri-sx-generator/test/siriSXUploadToDynamo.ts (hunk 3 above)
Comment (lj-kpmg) [reply packages/siri-sx-generator/test/siriSXUploadToDynamo.ts:7]: I think @alvak1 is referring to the script in the scripts folder which is a convenience wrapper around the npm command, it probably just needs updating to something like
```bash
#!/usr/bin/env bash

### Get the project root directory
cdd_root_folder_path=$(git rev-parse --show-toplevel)

stage_name=$(cat ${cdd_root_folder_path}/.sst/stage)
num_disruptions=$1

cd $cdd_root_folder_path/packages/siri-sx-generator

npm run dynamo-publish-json $stage_name $num_disruptions
```

File: site/pages/dashboard.page.tsx [hunk 4]
```diff
@@ -43,8 +44,46 @@ const formatDisruptionsIntoRows = (disruptions: DashboardDisruption[]) => {
     });
 };
 
+const sortEarliestDate = (firstDate: string, secondDate: string) => {
```
Comment (lj-kpmg) [inline site/pages/dashboard.page.tsx:47]:
```suggestion
const sortEarliestDate = (firstDate: string, secondDate: string) =>  
    getDate(firstDate).isBefore(getDate(secondDate)) ? -1 : 1;
```

File: site/components/PageNumbers.tsx [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+
+        for (let count = 0; count < numberOfPages; count++) {
+            pageNumbers.push(
+                <li
+                    key={`page-${count + 1}`}
+                    className={`govuk-pagination__item${
+                        count + 1 === currentPage ? " govuk-pagination__item--current" : ""
+                    }`}
+                >
+                    <a
```
Comment (lj-kpmg) [inline site/components/PageNumbers.tsx:26]: Next recommends using the <Link> component over <a> tags, which has the added benefit of being able to remove the eslint disable

File: site/pages/dashboard.page.tsx [hunk 6]
```diff
@@ -84,26 +131,41 @@ const Dashboard = ({ liveDisruptions, upcomingDisruptions }: DashboardProps): Re
...
-                                columns={["ID", "Summary", "Affected dates"]}
-                                rows={formatDisruptionsIntoRows(liveDisruptions)}
-                            />
+                            <>
+                                <Table
+                                    caption="Live disruptions"
+                                    columns={["ID", "Summary", "Affected dates"]}
+                                    rows={formatDisruptionsIntoRows(liveDisruptionsToDisplay)}
+                                />
+                                <PageNumbers
```
Comment (lj-kpmg) [inline site/pages/dashboard.page.tsx:141]: The gov guidance is to only show pagination if the number of pages is greater than 1 so we could put this behind a check on `numberOfLiveDisruptionsPages`

File: site/pages/dashboard.page.tsx [hunk 7]
```diff
@@ -84,26 +131,41 @@ const Dashboard = ({ liveDisruptions, upcomingDisruptions }: DashboardProps): Re
...
-                                columns={["ID", "Summary", "Affected dates"]}
-                                rows={formatDisruptionsIntoRows(upcomingDisruptions)}
-                            />
+                            <>
+                                <Table
+                                    caption="Upcoming disruptions"
+                                    columns={["ID", "Summary", "Affected dates"]}
+                                    rows={formatDisruptionsIntoRows(upcomingDisruptionsToDisplay)}
+                                />
+                                <PageNumbers
```
Comment (lj-kpmg) [inline site/pages/dashboard.page.tsx:158]: Same here

File: site/pages/dashboard.page.tsx [hunk 8]
```diff
@@ -43,8 +44,47 @@ const formatDisruptionsIntoRows = (disruptions: DashboardDisruption[]) => {
     });
 };
 
+const sortEarliestDate = (firstDate: string, secondDate: string) =>  
+    getDate(firstDate).isBefore(getDate(secondDate)) ? -1 : 1;
+    if (getDate(firstDate).isBefore(getDate(secondDate))) {
```
Comment (lj-kpmg) [inline site/pages/dashboard.page.tsx:49]: This may have gone awry

File: packages/siri-sx-generator/test/siriSXUploadToDynamo.ts (hunk 3 above)
Comment (RobCattonKPMG) [reply packages/siri-sx-generator/test/siriSXUploadToDynamo.ts:7]: Done

File: site/components/PageNumbers.tsx (hunk 5 above)
Comment (RobCattonKPMG) [reply site/components/PageNumbers.tsx:26]: Done

File: site/pages/dashboard.page.tsx (hunk 8 above)
Comment (RobCattonKPMG) [reply site/pages/dashboard.page.tsx]: Yeah it merged very strangely - fixed.

**Review by lj-kpmg — APPROVED** [review]

**Pull request merged**